x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_readonly/fields/kql_query/inline_kql_query.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiDescriptionList } from '@elastic/eui';
+import type { EuiDescriptionListProps } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+import { DataSourceType } from '../../../../../../../../../common/api/detection_engine';
+import type {
+  DiffableAllFields,
+  InlineKqlQuery,
+} from '../../../../../../../../../common/api/detection_engine';
+import { Query, Filters } from '../../../../rule_definition_section';
+import * as ruleDetailsI18n from '../../../../translations';
+import * as descriptionStepI18n from '../../../../../../../rule_creation_ui/components/description_step/translations';
+import { getQueryLanguageLabel } from '../../../../helpers';
+
+const defaultI18nLabels = {
+  query: descriptionStepI18n.QUERY_LABEL,
+  language: ruleDetailsI18n.QUERY_LANGUAGE_LABEL,
+  filters: descriptionStepI18n.FILTERS_LABEL,
+};
+
+interface InlineQueryProps {
+  kqlQuery: InlineKqlQuery;
+  dataSource?: DiffableAllFields['data_source'];
+  i18nLabels?: {
+    query: string;
+    language: string;
+    filters: string;
+  };
+}
+
+export function InlineKqlQueryReadOnly({
+  kqlQuery,
+  dataSource,
+  i18nLabels = defaultI18nLabels,
+}: InlineQueryProps) {
+  const listItems: EuiDescriptionListProps['listItems'] = [
+    {
+      title: i18nLabels.query,
+      description: <Query query={kqlQuery.query} />,
+    },
+    {
+      title: i18nLabels.language,
+      description: getQueryLanguageLabel(kqlQuery.language),
+    },
+  ];
+
+  if (kqlQuery.filters.length > 0 && dataSource) {
+    const index =
+      dataSource.type === DataSourceType.index_patterns ? dataSource.index_patterns : undefined;
+
+    const dataViewId =
+      dataSource.type === DataSourceType.data_view ? dataSource.data_view_id : undefined;
+
+    listItems.push({
+      title: i18nLabels.filters,
+      description: (
+        <Filters filters={kqlQuery.filters as Filter[]} index={index} dataViewId={dataViewId} />
```

Comment:
Just to follow up from the conversation a couple weeks ago, the `Filters` component needs either `index` or `dataViewId` and we can't display it when neither exists (`data_source` is undefined) just because it needs it for validation? Will we still be selecting the "fallback" `data_source` objects in whatever component that uses this?

Reply:
Yes, `data_source` is needed mostly for validation and also some other smaller things in the `Filter` component.

But you brought up a good point. I've taken a deeper look at the backend code and it turned out that it's entirely possible that a rule might not have both `index` and `data_view_id`. In such case, when executing a rule we [fall back](https://github.com/nikitaindik/kibana/blob/0bead120f62732302f223c274dbdfc47b3194200/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/get_input_output_index.ts#L88) to a default index pattern. I've added a similar fallback in this [commit](https://github.com/nikitaindik/kibana/commit/cc068bffbba74bc4738d1917d43e4b12c4c69d09). So now even if `data_source` is `undefined`, filters would still render with a fallback index pattern.